src/Qir/Execution/QirCommandLineTool/Program.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Help;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Qir.Tools;
+
+namespace CommandLineCompiler
+{
+    class Program
+    {
+        private static async Task<int> Main(string[] args)
+        {
+            var buildCommand = CreateBuildCommand();
+
+            var root = new RootCommand() { buildCommand };
+            SetSubCommandAsDefault(root, buildCommand);
+            root.Description = "Command-line tool for processing QIR DLL files.";
+            root.TreatUnmatchedTokensAsErrors = true;
+
+            Console.OutputEncoding = Encoding.UTF8;
+            return await new CommandLineBuilder(root)
+                .UseDefaults()
+                .UseHelpBuilder(context => new QsHelpBuilder(context.Console))
+                .Build()
+                .InvokeAsync(args);
+        }
+
+        /// <summary>
+        /// Creates the Build command for the command line compiler, which is used to
+        /// compile the executables from a given QIR DLL.
+        /// </summary>
+        /// <returns>The Build command.</returns>
+        private static Command CreateBuildCommand()
+        {
+            var buildCommand = new Command("build", "(default) Build the executables from a QIR DLL.")
+            {
+                Handler = CommandHandler.Create((BuildOptions settings) =>
+                {
+                    if (settings.LibraryDirectories.Length < 1)
+                    {
+                        throw new ArgumentException("The '--libraryDirectories' option requires at least one argument.");
+                    }
+                    if (settings.IncludeDirectories.Length < 1)
+                    {
+                        throw new ArgumentException("The '--includeDirectories' option requires at least one argument.");
+                    }
+                    return QirTools.BuildFromQSharpDll(settings.QsharpDll, settings.LibraryDirectories[0], settings.IncludeDirectories[0], settings.ExecutablesDirectory);
+                })
+            };
+            buildCommand.TreatUnmatchedTokensAsErrors = true;
+
+            Option<FileInfo> qsharpDllOption = new Option<FileInfo>(
```

Comment:
Given that `qsharpDllOption` is not referred to later, you may be able to simply the code by passing it directly to `AddOption` instead of saving it to a variable.

Reply:
done